### ISSUE TITLE
False positive for `unresolved-reference`

### ISSUE THREAD

**Issue by JasonGrace2282** [astral-sh/ty#3157]
### Summary

The following claims that `PULL_SUMS` is used without being defined, which is not the case.
```py
PULL_SUMS: "list[float]" = []

def bandit(foo: int) -> None:
    global PULL_SUMS

    if foo == 0:
        # commenting this out fixes it
        PULL_SUMS = []
        return

    def bar(arm):
        return PULL_SUMS[arm]
```

### Version

ty 0.0.26
